community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogEntryWriterV1.java

```diff
@@ -86,11 +86,10 @@ public void serialize( TransactionRepresentation tx ) throws IOException
     }
 
     @Override
-    public void writeCheckPointEntry( long transactionId, LogPosition logPosition ) throws IOException
+    public void writeCheckPointEntry( LogPosition logPosition ) throws IOException
     {
         writeLogEntryHeader( CHECK_POINT );
-        channel.putLong( transactionId ).
-                putLong( logPosition.getLogVersion() ).
+        channel.putLong( logPosition.getLogVersion() ).
```

Comment:
The idea is to save in the check point the log position where to start recovery from. Two benefits:
1) when you find the latest check point entry you know where to jump directly without searching for the commit entry
2) having the log version of the transaction referred by the latest check point helps in making sure that we prune the logs up to that version - 1

Note that potentially you might have the check point and the transaction in different log versions.